ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/FlexibleUtil.java

```diff
@@ -40,11 +42,53 @@ public static boolean isGroupAdditionTransaction(final PrivateTransaction privat
   }
 
   public static List<String> getParticipantsFromParameter(final Bytes input) {
+    final int numberOfParticipants;
+    try {
+      numberOfParticipants = input.slice(4 + 32, 32).toBigInteger().intValue();
+    } catch (final Exception exception) {
+      return Collections.emptyList();
+    }
+    if (numberOfParticipants == 0) return Collections.emptyList();
+    // Method selector + offset +  number of participants + (offset * number of participants)
+    final Bytes mungedParticipants = input.slice(4 + 32 + 32 + (32 * numberOfParticipants));
+
+    return getParticipantsFromMungedParticipants(mungedParticipants, numberOfParticipants);
+  }
+
+  public static List<String> decodeList(final Bytes rlpEncodedList) {
+    // first 32 bytes is dynamic list offset
+    if (rlpEncodedList.size() < 64) return new ArrayList<>();
+    // Bytes uses a byte[] for the content which can only have up to Integer.MAX_VALUE-5 elements
+    final int lengthOfList =
+        UInt256.fromBytes(rlpEncodedList.slice(32, 32)).toInt(); // length of list
+    if (lengthOfList == 0 || rlpEncodedList.size() < 64 + lengthOfList * 32) {
+      return new ArrayList<>();
+    }
+
+    final Bytes mungedParticipants = rlpEncodedList.slice(32 + 32 + (32 * lengthOfList));
+
+    return getParticipantsFromMungedParticipants(mungedParticipants, lengthOfList);
+  }
+
+  private static List<String> getParticipantsFromMungedParticipants(
+      final Bytes mungedParticipants, final int numberOfParticipants) {
     final List<String> participants = new ArrayList<>();
-    final Bytes mungedParticipants = input.slice(4 + 32 + 32);
-    for (int i = 0; i <= mungedParticipants.size() - 32; i += 32) {
-      participants.add(mungedParticipants.slice(i, 32).toBase64String());
+    // The participant value is enclosed in the closest multiple of 32 (for instance, 91 would be
+    // enclosed in 96)
+    final int sliceSize = mungedParticipants.size() / numberOfParticipants;
+    // All the participants have to have the same size, so it is enough to check the first one
+    final int participantSize;
+    try {
+      participantSize = mungedParticipants.slice(0, 32).toBigInteger().intValue();
+    } catch (final Exception exception) {
```

Comment:
when would this throw an exception?

Reply:
I have been reviewing this part and it was related to the tests `FlexiblePrivacyPrecompiledContractTest#testPrivateFromNotMemberOfGroup` and `FlexiblePrivacyPrecompiledContractTest#testInvalidResponseToMembershipQuery`, I have had to adapt the encoding for them so they break for the right reason. Updated now.